core/eth2signeddata.go

```diff
@@ -27,25 +29,34 @@ var (
 	_ Eth2SignedData = SyncCommitteeSelection{}
 )
 
-// VerifyEth2SignedData verifies signature associated with given Eth2SignedData.
+// VerifyEth2SignedData verifies signatures associated with the given Eth2SignedData.
 func VerifyEth2SignedData(ctx context.Context, eth2Cl eth2wrap.Client, data Eth2SignedData, pubkey tbls.PublicKey) error {
 	epoch, err := data.Epoch(ctx, eth2Cl)
 	if err != nil {
 		return err
 	}
 
-	sigRoot, err := data.MessageRoot()
+	sigs := data.Signatures()
+	domainNames := data.DomainNames()
+	msgRoots, err := data.MessageRoots()
 	if err != nil {
 		return err
 	}
 
-	return signing.Verify(ctx, eth2Cl, data.DomainName(), epoch, sigRoot, data.Signature().ToETH2(), pubkey)
+	for i, sig := range sigs {
+		err = signing.Verify(ctx, eth2Cl, domainNames[i], epoch, msgRoots[i], sig.ToETH2(), pubkey)
```

Comment:
If `domainNames` or `msgRoots` don't contain the `i`-th element, this code will panic.

Please add some bounds checks before falling into this for loop.

Reply:
yes correct, added checks before this for loop is executed:
```go
	if len(domainNames) != len(msgRoots) {
		return errors.New("mismatching lengths", z.Int("domain_names", len(domainNames)), z.Int("message_roots", len(msgRoots)))
	}
	if len(domainNames) != len(sigs) {
		return errors.New("mismatching lengths", z.Int("domain_names", len(domainNames)), z.Int("signatures", len(sigs)))
	}
```